backend/src/models/user.rs

```diff
@@ -94,23 +109,95 @@ impl users::Model {
     }
 
     pub async fn create_with_password(
-        database: &DatabaseConnection,
-        uuid: SeaOrmUuid,
+        app_state: &web::Data<app_state::AppState>,
         password_hash: String,
         credentials: &RegisterModel,
     ) -> Result<(), Error> {
-        users::Model::assert_is_unique(database, &credentials.email, &credentials.name).await?;
+        users::Model::assert_is_unique(&app_state.database, &credentials.email, &credentials.name)
+            .await?;
 
-        users::ActiveModel {
-            id: Set(uuid),
-            username: Set(credentials.name.clone()),
+        let user_info = UserInformation {
+            name: credentials.name.clone(),
+            email: credentials.email.clone(),
+            password_hash: password_hash.clone(),
+        };
+
+        let confirmation_code = uuid_gen::new_v4().to_string();
+
+        email_confirmation::Entity::insert(email_confirmation::ActiveModel {
             email: Set(credentials.email.clone()),
-            password: Set(Some(password_hash)),
+            user_info: Set(serde_json::to_value(user_info)?),
+            code: Set(confirmation_code.clone()),
+            #[allow(deprecated)]
+            expiration_date: Set(chrono::NaiveDateTime::from_timestamp(
+                (chrono::Utc::now() + Duration::minutes(30)).timestamp(),
+                0,
+            )),
+        })
+        .exec(&app_state.database)
+        .await?;
+
+        let mut confirmation_link = "".to_string();
+        confirmation_link.push_str(&EnvConfig::get().email_confirm_backend_url.clone());
+        confirmation_link.push('/');
+        confirmation_link.push_str(&confirmation_code);
+
+        Email {
+            sender: (
+                Some("Hack4Krak Authentication".to_string()),
+                "auth@hack4krak.pl".to_string(),
+            ),
+            recipients: vec![credentials.email.clone()],
+            subject: "Potwierdzenie rejestracji".to_string(),
+            template: EmailTemplate::EmailConfirmation,
+            placeholders: Some(vec![
+                ("user".to_string(), credentials.name.clone()),
+                ("link".to_string(), confirmation_link.to_string()),
+            ]),
+        }
+        .send(app_state)
+        .await?;
+
+        Ok(())
+    }
+
+    pub async fn confirm_email(
```

Comment:
It doesn't fit user model. Move it to a new file where you implement `email_confirmation::Entity`